src/lib/panic.c

```diff
@@ -70,18 +71,19 @@ void panic_rewind(uint32_t p, uint32_t stack_rewind_frames,
 
 	count = MAILBOX_EXCEPTION_SIZE -
 		(size_t)(ext_offset - mailbox_get_exception_base());
-	/* dump stack frames */
-	p = dump_stack(p, ext_offset, stack_rewind_frames, count);
 
 	/* flush last trace messages */
 #if CONFIG_TRACE
 	trace_flush();
 #endif
 
+	/* dump stack frames */
```

Comment:
Why dump the stack after the trace_flush?
Will the trace flush influence the stack we want to dump?

Reply:
The DSP state dumped should be consistent. This implementation for Xtensa dumps top frames stored in AR register file inside the `arch_dump_regs_a` function and this is the place where second part of the stack stored in memory should be dumped as well, note that arch_dump_regs_a is call8-ed. This simple move allows to record any in-memory stack change caused by spill(s) when calling the `trace_flush()` or any other function calls in between. Still misses any impact of the last call.